packages/installer/src/executors/run-command-executor.tsx

```diff
@@ -0,0 +1,123 @@
+import {spawn} from "cross-spawn"
+import {Box, Text} from "ink"
+import Spinner from "ink-spinner"
+import * as React from "react"
+import {Newline} from "../components/newline"
+import {RecipeCLIArgs} from "../types"
+import {useEnterToContinue} from "../utils/use-enter-to-continue"
+import {useUserInput} from "../utils/use-user-input"
+import {Executor, ExecutorConfig, getExecutorArgument} from "./executor"
+
+export interface CliCommand {
+  command: string
+  commandArgs?: string[]
+}
+
+export interface Config extends ExecutorConfig {
+  command: CliCommand
+}
```

Comment:
What about this way of passing the command?

```ts
export type CliCommand = string | [string, ...string[]]

export interface Config extends ExecutorConfig {
  command: CliCommand
}
```

And you can split the command into arguments with this regex:

```ts
export async function executeCommand(command: CliCommand) {
  // from https://stackoverflow.com/a/43766456/9950655
  const argsRegex = /("[^"\\]*(?:\\[\S\s][^"\\]*)*"|'[^'\\]*(?:\\[\S\s][^'\\]*)*'|\/[^/\\]*(?:\\[\S\s][^/\\]*)*\/[gimy]*(?=\s|$)|(?:\\\s|\S)+)/g
  const parsedCommand: string[] = Array.isArray(command) ? command : command.match(argsRegex) || []

  if (parsedCommand.length === 0) {
    throw new Error("The command is too short: " + JSON.stringify(command))
  }

  await new Promise((resolve) => {
    const cp = spawn(parsedCommand[0], parsedCommand.slice(1), {
      stdio: ["inherit", "pipe", "pipe"],
    })
    cp.on("exit", resolve)
  })
}
```

Reply:
needed a minute to understand what the snippet is doing - but yeah, looks good to me.

I updated the code and the unit tests - seems to work as expected.

Reply:
One last thing: the syntax of `[string, ...string[]]` is a trick for TypeScript so you don't get an empty array:

```ts
type NormalArray = string[]

const test1: NormalArray = [] // valid
const test2: NormalArray = ["ls"] // valid
const test3: NormalArray = ["ls", "-a"] // valid
const test4: NormalArray = ["ls", "-a", "-l"] // valid

type TrickArray = [string, ...string[]]

const test5: TrickArray = [] // error
const test6: TrickArray = ["ls"] // valid
const test7: TrickArray = ["ls", "-a"] // valid
const test8: TrickArray = ["ls", "-a", "-l"] // valid
```

So it isn't necessary to write the command as `["ls", ...["-a", "-l"]]`, just `["ls", "-a", "-l"]` will work

Reply:
i have updated the test script to test valid commands and also the thrown error to ensure everything is working as expected.